apps/mdreader/app/hooks/useFolders.ts

```diff
@@ -1,32 +1,53 @@
-import useSWR from 'swr';
 import { useMemo } from 'react';
+import useSWR from 'swr';
 
-import { folderService } from '../services/folder';
+import { FolderService } from '~/services/FolderService';
 import useSupabase from './useSupabase';
 
 type Props = {
   userId: string;
   folderId: number | null;
 };
 
-export const useFolder = (folderId?: number) => {
+export const useFolderService = () => {
   const client = useSupabase();
-  const service = useMemo(() => folderService(client), [client]);
+
+  const service = useMemo(
+    () => new FolderService({ supabase: client }),
+    [client]
+  );
+
+  return service;
+};
+
+export const useFolder = (folderId?: number) => {
+  const service = useFolderService();
 
   return useSWR({ folderId, key: 'folder' }, folderId ? service.getOne : null);
 };
 
-export const useFolders = (props: Props) => {
-  const client = useSupabase();
-  const service = useMemo(() => folderService(client), [client]);
-
-  return useSWR(
-    props.userId
-      ? {
-          ...props,
-          key: 'folders',
-        }
-      : null,
-    service.getAll
+export const useFolders = ({ folderId, userId }: Props) => {
+  const service = useFolderService();
+
+  return useSWR(userId ? `/folders/${userId}/${folderId ?? 0}` : null, () =>
```

Comment:
**Suggestion:** Consider using a more specific name for the 'useSWR' hook. Instead of 'useSWR', it could be 'useFoldersSWR' to make it clear what data it fetches.
```suggestion
  return useFoldersSWR(userId ? `/folders/${userId}/${folderId ?? 0}` : null, () =>
```